ansys/mapdl/core/logging.py

```diff
@@ -0,0 +1,499 @@
+
+    def __init__(self, logger, extra=None):
+        self.logger = logger
+        self.extra = extra
+        self._file_handler = logger._file_handler
+        self._std_out_handler = logger._std_out_handler
+
+    def process(self, msg, kwargs):
+        kwargs['extra'] = {}
+        # This are the extra parameters sent to log
+        kwargs['extra']['instance_name'] = self.extra['name'] # here self.extra is the argument pass to the ``PyAnsys
+        return msg, kwargs
+
+    def log_to_file(self, filename=FILE_NAME, level=LOG_LEVEL):
+        """
+        Add file handler to logger.
+
+        Parameters
+        ----------
+        filename : str, optional
+            Name of the file where the logs are recorded. By default FILE_NAME
+        level : str, optional
+            Level of logging. E.x. 'DEBUG'. By default LOG_LEVEL
+        """
+
+        self.logger = add_file_handler(self.logger, filename=filename, level=level, write_headers=True)
+        self._file_handler = self.logger._file_handler
+
+    def log_to_stdout(self, level=LOG_LEVEL):
+        """
+        Add standard output handler to the logger.
+
+        Parameters
+        ----------
+        level : str, optional
+            Level of logging record. By default LOG_LEVEL
+        """
+
+        self.logger = add_stdout_handler(self.logger, level=level)
+        self._std_out_handler = self.logger._std_out_handler
+
+    ## Copy functions
+    add_stdout_handler = log_to_stdout
+    add_file_handler = log_to_file
+
+
+class PyAnsysPercentStyle(logging.PercentStyle):
+
+    def __init__(self, fmt, *, defaults=None):
+        self._fmt = fmt or self.default_format
+        self._defaults = defaults
+
+    def _format(self, record):
+        defaults = self._defaults
+        if defaults:
+            values = defaults | record.__dict__
+        else:
+            values = record.__dict__
+
+        # We can do here any changes we want in record, for example adding a key.
+
+        # We could create an if here if we want conditional formatting, and even
+        # change the record.__dict__.
+        # Since now we don't want to create conditional fields, it is fine to keep
+        # the same MSG_FORMAT for all of them.
+        if 'instance_name' not in values.keys():  # For the case of logging exceptions to the logger.
+            values['instance_name'] = ''
+
+        return STDOUT_MSG_FORMAT % values
+
+
+class PyAnsysFormatter(logging.Formatter):
+    """Customized ``Formatter`` class used to overwrite the defaults format styles."""
+
+    def __init__(self, fmt=STDOUT_MSG_FORMAT, datefmt=None, style='%', validate=True, *, defaults=None):
+        super().__init__(fmt, datefmt, style, validate)
+        self._style = PyAnsysPercentStyle(fmt, defaults=defaults)  # overwriting
+
+
+class PyansysLogger():
+    """Logger used for each Pyansys logger.
+    This class allows you to add handler to a file or standard output.
+    Parameters
```

Comment:
Need whitespace between numpydoc sections.

```suggestion
    """Logger used for each Pyansys logger.

    This class allows you to add handler to a file or standard output.

        Parameters
```

Reply:
Done!